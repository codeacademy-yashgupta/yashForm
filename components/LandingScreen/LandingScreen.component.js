import React, { Component } from 'react';
import {
  StyleSheet, View,
} from 'react-native';
import Header from '../Header/Header.component';
import FormList from '../FormList/FormList.component';
import FloatingAddButton from '../FloatingAddButton/FloatingAddButton.component';
import { httpGet } from '../../utils/httpGet';
import { URL_FOR_GET_FORMS } from '../../constants';

export default class LandingScreen extends Component {
  state = {
    forms: [],
  }

  async componentDidMount() {
    const forms = await httpGet(URL_FOR_GET_FORMS);
    this.setState({ forms: forms.data });
  }

  render() {
    const { forms } = this.state;
    return (
      <View style={styles.container}>
        <Header isLandingPage="true" navigation={this.props.navigation} />
        <FormList forms={forms} navigation={this.props.navigation} />
        <FloatingAddButton navigation={this.props.navigation} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});
