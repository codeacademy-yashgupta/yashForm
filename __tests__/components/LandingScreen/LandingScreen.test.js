import React from 'react';
import renderer from 'react-test-renderer';
import LandingScreen from '../../../components/LandingScreen/LandingScreen.component';
import 'react-native';

describe('LandingScreen', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<LandingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
