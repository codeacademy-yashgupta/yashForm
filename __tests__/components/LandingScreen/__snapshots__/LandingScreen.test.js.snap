// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingScreen should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#3700B3",
        "flexDirection": "row",
        "height": 60,
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source=""
        style={
          Object {
            "marginTop": 12,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 20,
          "marginTop": 16,
        }
      }
    >
      Header
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../public/icons/baseline_more_vert_white_48pt_3x.png",
        }
      }
      style={
        Object {
          "height": 47,
          "marginTop": 6,
          "position": "absolute",
          "resizeMode": "contain",
          "right": -50,
        }
      }
    />
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#3700B3",
        "borderRadius": 50,
        "bottom": 0,
        "marginBottom": 30,
        "marginRight": 30,
        "padding": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../public/icons/baseline_add_white_48pt_3x.png",
          }
        }
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
  </View>
</View>
`;
